Automattic/woocommerce-payments
# Add a check on credit card being enabled to determine whether WCPay gateway is available for checkout

Fixes #3969 

#### Changes proposed in this Pull Request











#### Testing instructions

Classic checkout & Blocks checkout

- Visit the store, add a product to the cart and go to the checkout page: you should see the credit card payment method fields
- Go to Payment > Settings and disable the payment method `Credit card / debit card`
- Visit the store, add a product to the cart and go to the checkout page: you should not see the credit card payment method fields
- Go back to Payment > Settings and click on `Enable in your store` to enable UPE (add the `Bancontact` method)
- Switch to the General tab and set the currency of your store to US dollar
- Visit the store, add a product to the cart and go to the checkout page: you should not see the credit card payment method fields and you should also not see the UPE method you added earlier (since it is not compatible with the store's currency)
- Now, go back to wp-admin and change the currency of your store to EUR
- Visit the store, add a product to the cart and go to the checkout page: you should not see the credit card payment method fields BUT now you should see the `Bancontact` UPE method option. You should be able to place an order using whatever payment method is available to you (while credit card is not).
- Just to be sure, if you now enable again  the payment method `Credit card / debit card` in Payments > Settings, visit the store, add a product to the cart and go to the checkout page: you should see the credit card option along with the UPE one.

-------------------


**Post merge**
